cpp/benchmarks/join/generate_input_tables.cuh

```diff
@@ -0,0 +1,408 @@
+        lottery, lottery_size, devStates, num_states
+    );
+
+    CUDA_RT_CALL(cudaGetLastError());
+    CUDA_RT_CALL(cudaDeviceSynchronize());
+
+    CUDA_RT_CALL(cudaMemcpy(
+        build_tbl_sorted, build_tbl, build_tbl_size * sizeof(key_type), cudaMemcpyDeviceToDevice
+    ));
+
+    thrust::sort(thrust::device, build_tbl_sorted, build_tbl_sorted + build_tbl_size);
+
+    // Exclude keys used in build table from lottery
+    thrust::counting_iterator<key_type> first_lottery_elem(0);
+    thrust::counting_iterator<key_type> last_lottery_elem = first_lottery_elem + lottery_size;
+    key_type *lottery_end = thrust::set_difference(
+        thrust::device, first_lottery_elem, last_lottery_elem,
+        build_tbl_sorted, build_tbl_sorted + build_tbl_size, lottery
+    );
+
+    lottery_size = thrust::distance(lottery, lottery_end);
+
+    init_probe_tbl<key_type, size_type><<<num_sms * num_blocks_init_build_tbl, block_size>>>(
+        probe_tbl, probe_tbl_size, build_tbl, build_tbl_size,
+        lottery, lottery_size, selectivity, devStates, num_states
+    );
+
+    CUDA_RT_CALL(cudaGetLastError());
+    CUDA_RT_CALL(cudaDeviceSynchronize());
+
+    if (lottery_in_device_memory) {
+        CUDA_RT_CALL(cudaFree(lottery));
+    } else {
+        CUDA_RT_CALL(cudaFreeHost(lottery));
+    }
+
+    CUDA_RT_CALL(cudaFree(build_tbl_sorted));
+    CUDA_RT_CALL(cudaFree(devStates));
+}
+
+
+template<typename key_type, typename size_type>
+__global__ void linear_sequence(key_type* tbl, const size_type size)
+{
+  for (size_type i = threadIdx.x + blockDim.x * blockIdx.x; i < size; i += blockDim.x * gridDim.x)
+    tbl[i] = i;
+}
+
+
+/**
+ * Generate a build table and a probe table for testing join performance.
+ *
+ * Both the build table and the probe table have two columns. The first column is the key column,
+ * with datatype KEY_T. The second column is the payload column, with datatype PAYLOAD_T. Both the
+ * arguments build_table and probe_table do not need to be preallocated. It is the caller's
+ * responsibility to free memory of build_table and probe_table allocated by this function.
+ *
+ * @param[out] build_table         The build table to generate.
+ * @param[in] build_table_size     The number of rows in the build table.
+ * @param[out] probe_table         The probe table to generate.
+ * @param[in] probe_table_size     The number of rows in the probe table.
+ * @param[in] selectivity          Propability with which an element of the probe table is present in
+ *                                 the build table.
+ * @param[in] rand_max             Maximum random number to generate, i.e., random numbers are
+ *                                 integers from [0, rand_max].
+ * @param[in] uniq_build_tbl_keys  If each key in the build table should appear exactly once.
+ */
+template<typename KEY_T, typename PAYLOAD_T>
+void generate_build_probe_tables(std::vector<gdf_column *> &build_table,
```

Comment:
Use a `cudf::table` instead of a `vector<gdf_column*>`

Reply:
I removed this helper function because
1. If we put the columns generated by column_wrapper into a cudf::table, I think we need to keep track of these wrappers so that we can call destructor, in addition to keep track of the table itself. This makes using a helper function less convenient.
2. After using column_wrapper, the logic is much simpler.